src/manifest_parser.cc

```diff
@@ -146,8 +146,10 @@ bool ManifestParser::ParseRule(string* err) {
   while (lexer_.PeekToken(Lexer::INDENT)) {
     string key;
     EvalString value;
-    if (!ParseLet(&key, &value, err))
+    if (!ParseLet(&key, &value, err)) {
+      delete rule;
```

Comment:
Why not just replace it with `std::unique_ptr`?

Reply:
Seems like overkill, but maybe?